test/e2e/storage/persistent_volumes-local.go

```diff
@@ -266,6 +292,57 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 			framework.DeletePodOrFail(config.client, config.ns, pod2.Name)
 		})
 	})
+
+	Context("when using local volume provisioner", func() {
+		var (
+			volumePath string
+		)
+
+		BeforeEach(func() {
+			setupLocalVolumeProvisioner(config, node0)
+			volumePath = path.Join(
+				hostBase, discoveryDir, fmt.Sprintf("vol-%v", string(uuid.NewUUID())))
+		})
+
+		AfterEach(func() {
+			cleanupLocalVolumeProvisioner(config, node0, volumePath)
+		})
+
+		It("should create and recreate local persistent volume", func() {
+			By("Creating bootstrapper pod to start provisioner daemonset")
+			createBootstrapperPod(config)
+			kind := schema.GroupKind{"extensions", "DaemonSet"}
+			framework.WaitForControlledPodsRunning(config.client, config.ns, daemonSetName, kind)
+
+			By("Creating a directory under discovery path")
+			framework.Logf("creating local volume under path %q", volumePath)
+			mkdirCmd := fmt.Sprintf("mkdir %v", volumePath)
+			runLocalUtil(config, node0.Name, mkdirCmd, hostBase)
+			oldPV, err := waitForLocalPersistentVolume(config.client, volumePath)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Create a persistent volume claim for local volume: the above volume will be bound.
+			By("Creating a persistent volume claim")
+			claim, err := config.client.Core().PersistentVolumeClaims(config.ns).Create(newLocalClaim(config))
+			Expect(err).NotTo(HaveOccurred())
+			err = framework.WaitForPersistentVolumeClaimPhase(
+				v1.ClaimBound, config.client, claim.Namespace, claim.Name, framework.Poll, 1*time.Minute)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Delete the persistent volume claim: file will be cleaned up and volume be re-created.
+			By("Deleting the persistent volume claim to clean up persistent volume and re-create one")
+			writeCmd, readCmd := createWriteAndReadCmds(volumePath, testFile, testFileContent)
+			runLocalUtil(config, node0.Name, writeCmd, hostBase)
+			err = config.client.Core().PersistentVolumeClaims(claim.Namespace).Delete(claim.Name, &metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			newPV, err := waitForLocalPersistentVolume(config.client, volumePath)
+			Expect(newPV.UID).NotTo(Equal(oldPV.UID))
+			Expect(err).NotTo(HaveOccurred())
+			result, err := framework.NodeExec(node0.Name, readCmd)
```

Comment:
I didn't realize there was a NodeExec command.  Can we use it to also initially populate the local volumes?  Instead of the current method of creating a pod with hostpath?